plugins/gatsby-plugin-check-links/utils/check-nav-links.js

```diff
@@ -0,0 +1,61 @@
+const fs = require('fs');
+const yaml = require('js-yaml');
+
+const NAV_DIR = 'src/nav';
+const getPageResponse = require('./get-page-response');
+
+const extractLinks = (data, links = []) => {
+  if (!data.pages) {
+    return data.path;
+  }
+
+  const linksAtLevel = data.pages.flatMap((page) => extractLinks(page, links));
+
+  return data.path ? [...linksAtLevel, data.path] : linksAtLevel;
+};
+
+const getLinksFromNav = (filepath) => {
+  try {
+    const file = fs.readFileSync(filepath, 'utf8');
+    const data = yaml.load(file);
+
+    return { links: [...new Set(extractLinks(data))], filepath };
+  } catch (e) {
+    console.error(`Unable to fetch ${filepath}:\n${e}`);
+  }
+};
+
+const codeRequests = async (nav) => {
```

Comment:
If we find that this takes a while to run, we might want to consider running these requests in batches in parallel.